src/main/java/com/thealgorithms/sorts/RadixSort.java

```diff
@@ -2,61 +2,58 @@
 
 import java.util.Arrays;
 
-final class RadixSort {
+/**
+ * This class provides an implementation of the radix sort algorithm.
+ * It sorts an array of integers in increasing order.
+ */
+public final class RadixSort {
     private RadixSort() {
     }
 
-    private static int getMax(int[] arr, int n) {
-        int mx = arr[0];
-        for (int i = 1; i < n; i++) {
-            if (arr[i] > mx) {
-                mx = arr[i];
-            }
+    /**
+     * Sorts an array of integers using the radix sort algorithm.
+     *
+     * @param array the array to be sorted
+     * @return the sorted array
+     */
+    public static int[] sort(int[] array) {
+        if (array.length == 0) {
+            return array;
         }
-        return mx;
+
+        final int max = Arrays.stream(array).max().getAsInt();
+        for (int exp = 1; max / exp > 0; exp *= 10) {
+            countingSortByDigit(array, exp);
+        }
+
+        return array;
     }
 
-    private static void countSort(int[] arr, int n, int exp) {
-        int[] output = new int[n];
-        int i;
+    /**
+     * A utility method to perform counting sort of array[] according to the digit represented by exp.
+     *
+     * @param array the array to be sorted
+     * @param exp   the exponent representing the current digit position
+     */
+    private static void countingSortByDigit(int[] array, int exp) {
+        int[] output = new int[array.length];
         int[] count = new int[10];
-        Arrays.fill(count, 0);
 
-        for (i = 0; i < n; i++) {
-            count[(arr[i] / exp) % 10]++;
+        for (int i = 0; i < array.length; i++) {
+            count[(array[i] / exp) % 10]++;
```

Comment:
What do you think about introducing a method like `private static int getDigit(int number, int position)`?

Reply:
Thanks, need to add. This is a complex expression.